tests/FunctionalJavascript/TranslationSynchronisationFieldWidgetTest.php

```diff
@@ -0,0 +1,164 @@
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_translation\FunctionalJavascript;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Entity\Entity\EntityFormDisplay;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\language\Entity\ConfigurableLanguage;
+use Drupal\node\Entity\Node;
+
+/**
+ * Tests the Translation Synchronisation field widget.
+ */
+class TranslationSynchronisationFieldWidgetTest extends WebDriverTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'system',
+    'node',
+    'user',
+    'field',
+    'text',
+    'options',
+    'oe_translation',
+    'language',
+  ];
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+
+    $this->entityTypeManager->getStorage('node_type')->create([
+      'name' => 'Page',
+      'type' => 'page',
+    ])->save();
+
+    FieldStorageConfig::create([
+      'field_name' => 'translation_sync',
+      'entity_type' => 'node',
+      'type' => 'oe_translation_translation_sync',
+    ])->save();
+
+    FieldConfig::create([
+      'field_name' => 'translation_sync',
+      'entity_type' => 'node',
+      'bundle' => 'page',
+    ])->save();
+
+    ConfigurableLanguage::create(['id' => 'fr'])->save();
+    ConfigurableLanguage::create(['id' => 'it'])->save();
+
+    $entity_form_display = EntityFormDisplay::collectRenderDisplay(Node::create(['type' => 'page']), 'default');
+    $entity_form_display->setComponent('translation_sync', [
+      'weight' => 1,
+      'region' => 'content',
+      'type' => 'oe_translation_translation_sync_widget',
+      'settings' => [],
+      'third_party_settings' => [],
+    ]);
+    $entity_form_display->save();
+
+    $user = $this->drupalCreateUser([], NULL, TRUE);
+    $this->drupalLogin($user);
+  }
+
+  /**
+   * Tests the Translation Synchronisation field widget.
+   */
+  public function testTranslationSynchronisationFieldWidget(): void {
+    $this->drupalGet('/node/add/page');
+
+    $assert = $this->assertSession();
+    $assert->fieldExists('title[0][value]')->setValue('My page');
+
+    // Assert the select is correct.
+    $select = $assert->selectExists('translation_sync[0][type]');
+    $assert->optionExists('Select type', 'None');
+    $assert->optionExists('Select type', 'Manual');
+    $assert->optionExists('Select type', 'Automatic with minimum threshold');
+    $this->assertEquals('None', $select->find('xpath', "//option[@selected='selected']")->getText());
+
+    // Assert we don't see a configuration fieldset.
+    $this->assertFalse($this->getSession()->getPage()->find('css', '#edit-translation-sync-0-configuration')->isVisible());
```

Comment:
I would avoid using the ids to find the configuration. Why not use the selectExists method like before?

Reply:
It's fine, we assert the entire container cause thats what the states affects.